rholang/src/test/scala/coop/rchain/rholang/interpreter/ReplaySpec.scala

```diff
@@ -0,0 +1,123 @@
+package coop.rchain.rholang.interpreter
+
+import cats.effect.Resource
+import cats.syntax.all._
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.metrics
+import coop.rchain.metrics.{Metrics, NoopSpan, Span}
+import coop.rchain.rholang.Resources
+import coop.rchain.rholang.interpreter.accounting.utils.costLog
+import coop.rchain.rholang.interpreter.accounting.{_cost, Cost, CostAccounting}
+import coop.rchain.rspace.SoftCheckpoint
+import coop.rchain.shared.Log
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.duration._
+
+class ReplaySpec extends FlatSpec with Matchers {
+
+  it should "not fail with unused COMM events ???" in {
```

Comment:
Is this test passing now that you have ordered the produces by channel hash?

Reply:
@Isaac-DeFrain this is a new test that reveals a new bug when produces happens after consume e.g. `x!() | x!() | for(<- x; <- x) { 0 }`. If `x!()` happens after consume (or both sends), replay will save sends and it will not find COM event.

Ordering bug is more likely to happen than this new bug, so this test would detect ordering bug also.